### ISSUE TITLE
[Question] Avoid downloading MongDB for every package

### ISSUE THREAD

**Issue by vinyoliver** [shelfio/jest-mongodb#181]
Hi,

I'm facing the following problem, I've got several microservices structured in a way that every microservice is a project. Every microservice inherits from the base package which holds the commons dependencies used for every project. One of these dependencies is `jest-mongodb` and `mongodb` for unit testing;

The problem I am facing is that every microservice is downloading a mongoDB instead of downloading just once for the base package.json where it's defined. Any suggestion on how to solve this?

**Comment by sibelius** [comment]
Use `mongodb-memory-server-global` instead, it should fix for you

**Comment by qqilihq** [comment]
How would that work? `jest-mongodb` directly depends on `mongodb-memory-server`(non global), how can this be parametrized?

**Comment by sibelius** [comment]
you should open an issue there

**Comment by qqilihq** [comment]
So, to clarify: The suggestion [above](https://github.com/shelfio/jest-mongodb/issues/181#issuecomment-640603030) will not fix this.

`mongodb-memory-server` already provides **three** separate packages:

* `mongodb-memory-server` -- *“Auto-downloads version […] mongod binary on npm install to: node_modules/.cache/mongodb-binaries.”*
* `mongodb-memory-server-global` -- *“Auto-downloads […] mongod binary on npm install to: %HOME%/.cache/mongodb-binaries / ~/.cache/mongodb-binaries.”*
* `mongodb-memory-server-core` -- *“Does NOT auto-download mongod on npm install.”*

`jest-mongodb` should ideally have a mechanism to define on which of these three to depend (instead of forcing `mongodb-memory-server`).

A quick idea would be to declare the dependency to `mongodb-memory-server` from this package as optional or peerDependency, and let the user decide on which three options to depend (by pulling in the desired one). Would you be open to a MR in that direction, @vladgolubev ?

**Comment by SalvatorePreviti** [comment]
I am having issues as well with different versions of the package. Can mongodb-memory-server be set as peerDependency?

**Comment by hasezoey** [comment]
I know this is old, but there is a workaround directly in mongodb-memory-server (since 7.0.0):
Package `mongodb-memory-server`(modules cache) only sets the default value for [`PREFER_GLOBAL_PATH`](https://typegoose.github.io/mongodb-memory-server/docs/api/config-options#prefer_global_path) to `false`.
Whereas `-global`(homedir) packages only set the default value for [`DOWNLOAD_DIR`](https://typegoose.github.io/mongodb-memory-server/docs/api/config-options#download_dir) to the home directory.
This means that any overwrites (via package.json or environment variables) should still be respected.

TL;DR: in `mongodb-memory-server` package you can set [`PREFER_GLOBAL_PATH`](https://typegoose.github.io/mongodb-memory-server/docs/api/config-options#prefer_global_path) to `true` and it should download to the home directory like a `-global` package.
